### ISSUE TITLE
TypeError: can't pickle thread.lock objects

### ISSUE THREAD

**Issue by radiantone** [dask/distributed#2366]
Hi,
  When I attempt to distribute a few functions in dask/distributed from within my Flask app wrapped in Twisted, it works fine. But when I switch to Bjoern it produces a lock serialization error:

  File "/data/darren/anaconda2/lib/python2.7/site-packages/distributed/client.py", line 1919, in get
    resources=resources)
  File "/data/darren/anaconda2/lib/python2.7/site-packages/distributed/client.py", line 1882, in _graph_to_futures
    'tasks': valmap(dumps_task, dsk3),
  File "/data/darren/anaconda2/lib/python2.7/site-packages/toolz/dicttoolz.py", line 84, in valmap
    rv.update(zip(iterkeys(d), map(func, itervalues(d))))
  File "/data/darren/anaconda2/lib/python2.7/site-packages/distributed/worker.py", line 686, in dumps_task
    return {'function': dumps_function(task[0]),
  File "/data/darren/anaconda2/lib/python2.7/site-packages/distributed/worker.py", line 650, in dumps_function
    b = pickle.dumps(func)
  File "/data/darren/anaconda2/lib/python2.7/site-packages/distributed/protocol/pickle.py", line 51, in dumps
    return cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
  File "/data/darren/anaconda2/lib/python2.7/site-packages/cloudpickle/cloudpickle.py", line 900, in dumps
    cp.dump(obj)
...
...
TypeError: can't pickle thread.lock objects

Any clues?

Thanks in advance

**Comment by martindurant** [comment]
The message means that in one of the functions/tasks you are attempting to run with Dask, there is an object which cannot be safely passed between processes. The object may be a direct argument to your function, in a closure or even just an attribute of one of the objects you are dealing with. Since this happened when switching your app framework, it is likely something that is part of that framework.

You would be best off trying to build the smallest standalone version of your code that shows the problem before trying to debug.

**Comment by radiantone** [comment]
Yeah. I get that. But the funny thing is if it were something in my code it wouldnt work at all. Yet only when I use Bjoern I see this. So my only reason for posting this really (which is not assuming an issue with bjoern) is that some interaction in the code resulting from use of bjoern is exposing this problem. Not yet sure where or why yet.

**Comment by jcrist** [comment]
No idea, we're not familiar with Bjoern. Perhaps Bjoern monkey-patches something in the standard library. If you can provide a [reproducible example](http://blog.dask.org/2018/02/28/minimal-bug-reports) we'd be happy to take a look.
